packages/automation/README.md

```diff
@@ -0,0 +1,347 @@
+
+### States
+
+States represent different phases of your automation. Each state can:
+
+- Execute code when entered and/or exited
+- Configure PKPs and Capacity Credits for the machine
+- Run Lit Actions
+- Send blockchain transactions
+- Run custom code
+
+### Transitions
+
+Transitions define how the machine moves between states. They can be triggered automatically or by any combination of:
+
+- Blockchain events
+- Token balance changes
+- Timers and intervals
+- HTTP requests (polling)
+- Custom conditions
+
+### Listeners
+
+Listeners monitor various events and feed data to transitions:
+
+- EVMBlockListener: Monitors new blocks
+- EVMContractEventListener: Monitors EVM smart contract events
+- TimerListener: Triggers based on time
+- FetchListener: Polls an HTTP endpoint at regular intervals
+- IntervalListener: Runs a function at regular intervals
+
+## Basic Example
+
+Here's a simple example that mints a PKP, a Capacity Delegation NFT and then runs a Lit Action every hour:
+
+```typescript
+async function runLitActionInterval() {
+  const stateMachine = StateMachine.fromDefinition({
+    privateKey: '0xPRIVATE_KEY_WITH_LIT_TOKENS',
+    litNodeClient: {
+      litNetwork: 'datil-test',
+    },
+    litContracts: {
+      network: 'datil-test',
+    },
+    states: [
+      {
+        key: 'setPKP',
+        usePkp: {
+          mint: true,
+        },
+        transitions: [{ toState: 'setCapacityNFT' }],
+      },
+      {
+        key: 'setCapacityNFT',
+        useCapacityNFT: {
+          mint: true,
+          daysUntilUTCMidnightExpiration: 10,
+          requestPerSecond: 1,
+        },
+        transitions: [{ toState: 'runLitAction' }],
+      },
+      {
+        key: 'runLitAction',
+        litAction: {
+          code: `(async () => {
+              if (magicNumber >= 42) {
+                  LitActions.setResponse({ response:"The number is greater than or equal to 42!" });
+              } else {
+                  LitActions.setResponse({ response: "The number is less than 42!" });
+              }
+            })();`,
+          jsParams: {
+            magicNumber: Math.floor(Math.random() * 100),
+          },
+        },
+        transitions: [{ toState: 'cooldown' }],
+      },
+      {
+        key: 'cooldown',
+        transitions: [
+          {
+            toState: 'runLitAction',
+            timer: {
+              until: 1 * 60 * 60 * 1000,
+            },
+          },
+        ],
+      },
+    ],
+  });
+
+  // Start the machine at the desired state
+  await stateMachine.startMachine('setPKP');
+}
+
+runLitActionInterval().catch(console.error);
+```
+
+## Functional interface
+
+There care cases where such a declarative interface won't be enough for your use case. When that happens, the machines can also accept generic states, transitions and listeners where it is possible to write any logic.
+
+Here is an example that listens to Ethereum blocks looking one whose numbers ends in 0
```

Comment:
> Here is an example that listens to Ethereum blocks looking one whose numbers ends in 0

Here is an example that listens to Ethereum blocks, looking for those that end in '0'